src/lib/ng-package/entry-point/entry-point.spec.ts

```diff
@@ -0,0 +1,174 @@
+    const secondaryEntryPointBasePath = join(testSourcePath, secondaryEntryPointRelativeSourcePath);
+    secondaryData = {
+      moduleId: `@example/test-entry-point/${ secondaryEntryPointRelativeSourcePath }`,
+      primaryDestinationPath: primaryEntryPoint.destinationPath,
+      destinationPath: join(primaryEntryPoint.destinationPath, secondaryEntryPointRelativeSourcePath),
+    };
+    secondaryEntryPoint = new NgEntryPoint(packageJson, secondaryNgPackageJson, secondaryEntryPointBasePath, secondaryData);
+
+    expect(secondaryEntryPoint.entryFilePath).to.equal(`${secondaryEntryPointBasePath}/public-api.ts`);
+    expect(secondaryEntryPoint.isSecondaryEntryPoint).to.be.true;
+    expect(secondaryEntryPoint.destinationPath).to.equal(secondaryData.destinationPath);
+    expect(Object.keys(secondaryEntryPoint.destinationFiles).length).to.equal(6);
+    expect(secondaryEntryPoint.entryFile).to.equal('public-api.ts');
+    expect(secondaryEntryPoint.cssUrl).to.be.undefined;  // TODO: should default to 'inline'.
+    expect(secondaryEntryPoint.umdModuleIds).to.be.undefined;
+    expect(secondaryEntryPoint.flatModuleFile).to.equal('example-test-entry-point-extra');
+    expect(secondaryEntryPoint.styleIncludePaths).to.be.empty;
+    expect(secondaryEntryPoint.moduleId).to.equal('@example/test-entry-point/extra');
+    expect(secondaryEntryPoint.umdId).to.equal('example.test-entry-point.extra');
+    expect(secondaryEntryPoint.amdId).to.equal('@example/test-entry-point/extra');
+    expect(secondaryEntryPoint.sideEffects).to.be.false;
+  });
+
+  it(`check secondary entry point overrides`, () => {
+    packageJson = {
+      name: '@example/test-entry-point',
+      sideEffects: true,
+    };
+    primaryNgPackageJson = {
+      dest: testDestinationPath,
+      lib: {
+        entryFile: 'public-api.ts',
+      },
+    };
+    primaryEntryPoint = new NgEntryPoint(packageJson, primaryNgPackageJson, testSourcePath);
+    secondaryNgPackageJson = {
+      dest: testDestinationPath,
+      lib: {
+        entryFile: 'demo-api.ts',
+        cssUrl: 'none',
+        umdModuleIds: {
+          '@example/test-module/extra': 'example.testModuleExtra',
+        },
+        flatModuleFile: 'exampleTestEntryPointExtra',
+        styleIncludePaths: ['./style-includes'],
+        umdId: 'example.testEntryPointExtra',
+        amdId: '@example/testEntryPointExtra',
+      },
+    };
+    const secondaryEntryPointRelativeSourcePath = 'extra';
+    const secondaryEntryPointBasePath = join(testSourcePath, secondaryEntryPointRelativeSourcePath);
+    secondaryData = {
+      moduleId: `@example/test-entry-point/${ secondaryEntryPointRelativeSourcePath }`,
+      primaryDestinationPath: primaryEntryPoint.destinationPath,
+      destinationPath: join(primaryEntryPoint.destinationPath, secondaryEntryPointRelativeSourcePath),
+    };
+    secondaryEntryPoint = new NgEntryPoint(packageJson, secondaryNgPackageJson, secondaryEntryPointBasePath, secondaryData);
```

Comment:
In a future PR, I plan to replace this `secondaryData` with a reference to the parent entry point, which contains all the needed data, plus some extra information that will help with various things: like inheriting defined module IDs (#1674), providing a flag to change the generated `umdId` to follow the patterns from the angular core packages (camelCase instead of dash-case module IDs), etc.